### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by hoxbro**
Updates required dependencies:
Numpy 1.0 --> 1.21
Pandas 0.20.0 --> 1.3
Bokeh (moved up from recommended)

Updates recommended dependencies:
Removed notebook and IPython. 
Moved Bokeh up to required.

### CODE REVIEW

**Review by maximlt — COMMENTED** [review]
To make it easier to understand the consequences of this change:

```
dependencies = [
    "param >=2.0,<3.0",
    "numpy >=1.0",
    "pyviz_comms >=2.1",
    "panel >=1.0",
    "colorcet",
    "packaging",
    "pandas >=0.20.0",
]
recommended = ["ipython >=5.4.0", "notebook", "matplotlib >=3", "bokeh >=3.1", "plotly >=4.0"]
```

Compared with what's on *conda-forge*:

```
  run:
    - python >=3.9
    - param >=1.12.0,<3.0
    - panel >=1.0
    - numpy >=1.0
    - pandas >=0.20.0
    - matplotlib-base >=3.0
    - bokeh >=3.1
    - colorcet
    - pyviz_comms >=0.7.4
    - packaging
```

And *defaults*:

```
  run:
    - python
    - colorcet
    - numpy >=1.0
    - packaging
    - panel >=1.0
    - pandas >=0.20.0
    - param >=1.12.0,<3.0
    - pyviz_comms >=0.7.4
    - matplotlib-base >=3
```

- I'd vote for aligning the runtime dependencies defined on these three channels.
- I care less about the `pyviz` channel, in fact I hope we could retire it in some way, only `pyviz/label/dev` seems useful these days. We should certainly remove references to the `-c pyviz` channel in the docs. Yet, removing/updating the `recommended` packages is a breaking change for users of this channel, so I'd motivate a change there by the goal of aligning the deps across the channels, rather than the intent to gain a few seconds on the CI.

**Comment by hoxbro** [comment]
> goal of aligning the deps across the channels, rather than the intent to gain a few seconds on the CI.

Users are also affected by this. We pull in a big dependency that we don't use. I just noticed it is because of our CI. 

For alignment, I would propose these as required dependencies.

``` python
bokeh >=3.1
colorcet
numpy >=1.0
packaging
pandas >=0.20.0
panel >=1.0
param >=2.0,<3.0
pyviz_comms >=2.1
```

Maybe even bump the lower pin of pandas and numpy to something released in the last 3-5 years, or even remove them as they are so low it doesn't matter. 

(numpy 1.0 was released 25th October 2006, pandas 0.20.0 was released 5th May 2017)

File: pyproject.toml [hunk 1]
```diff
@@ -30,13 +30,14 @@ classifiers = [
...
-    "param >=2.0,<3.0",
-    "numpy >=1.0",
-    "pyviz_comms >=2.1",
-    "panel >=1.0",
+    "bokeh >=3.1",
     "colorcet",
+    "numpy >=1.21",
     "packaging",
-    "pandas >=0.20.0",
+    "pandas >=1.3",
```
Comment (hoxbro) [inline pyproject.toml:37]: Pandas and numpy versions were changed to the first minor version with a micro release after Python 3.10 was released.

**Review by maximlt — APPROVED** [review]
LGTM. You may want to list the versions bumped and packages removed from `recommended` in the original post, to make it easier to write the release notes.

**Review by philippjfr — APPROVED** [review]
I'm fully on board with this.

**Pull request merged**